crates/gfx/src/lib.rs

```diff
@@ -97,13 +97,13 @@ impl CameraUniform {
 #[derive(Debug, Copy, Clone)]
 pub struct Position {
     /// The vector-representation of the position.
-    pub vector: Vector3<f32>,
+    pub point: Point3<f32>,
```

Comment:
In some cases, in others not. The main reason I chose to replace it was because it even exists, so the [`cgmath` library authors thought it important to distinguish](https://docs.rs/cgmath/latest/cgmath/trait.EuclideanSpace.html#point-vector-distinction) between points in space (that are relative to the origin) and vectors (which are between two points). The distinction makes a lot of sense, and I believe the types being different prevent users from performing operations that don't make any sense. 

Using `Point3` means that in some places you need to convert it to a `Vector3`, but I think that tradeoff is worth it for the clarity.